docs/encyclopedia/workflow/workflow-execution/continue-as-new.mdx

```diff
@@ -13,41 +13,45 @@ tags:
   - Workflows
 ---
 
-This page discusses [Continue-As-New](#continue-as-new).
-
-A Workflow Execution may run for an arbitrarily long period of time, during which it could potentially generate more events than allowed by the Event History.
-The Continue-As-New feature offers a potential solution for this, since it enables the progress to continue in a subsequent Workflow Execution with a fresh Event History.
-You can repeat this as often as needed, which means that your Workflow can effectively run in perpetuity.
+This page discusses [Continue-As-New](#continue-as-new) and how to tell [when](#when) you would want to use it.
 
 ## What is Continue-As-New? {#continue-as-new}
 
-Continue-As-New is a mechanism by which the latest relevant state is passed to a new Workflow Execution, with a fresh Event History.
+Continue-As-New allows you to checkpoint your Workflow's state and start a fresh Workflow.
 
-:::caution
+There are two main problems this practice can prevent:
 
-As a precautionary measure, the Workflow Execution's Event History is limited to [51,200 Events](https://github.com/temporalio/temporal/blob/e3496b1c51bfaaae8142b78e4032cc791de8a76f/service/history/configs/config.go#L382) or [50 MB](https://github.com/temporalio/temporal/blob/e3496b1c51bfaaae8142b78e4032cc791de8a76f/service/history/configs/config.go#L380) and will warn you after 10,240 Events or 10 MB.
+- A Workflow Execution with a long, or large [Event History](/workflow-execution/event#event-history), such as one calling many activities, may bog down and have performance issues.
+  It could even generate more events than allowed by the [Event History limits](/workflow-execution/event#event-history-limits).
+- A Workflow Execution can hit [Workflow Versioning](/workflow-definition#workflow-versioning) problems if it started running on an older version of your code.
 
-:::
+Pass your latest relevant state into Continue-As-New.
+This checkpoints it and passes it to a new Execution in the [Execution Chain](/workflow-execution#workflow-execution-chain).
+This state is passed in through your Workflow's parameters.
+The parameters are typically optional and left unset by the original caller of the Workflow.
```

Comment:
I'd probably get rid of "This checkpoints it" because as a reader that would make me think that Temporal had a thing called "checkpointing" that I didn't know about.

I'm also tempted to add a sentence saying something like

> The end result is that you've created a new workflow with a fresh history, that picks up where your last one left off.